src/VenueSearchIterator.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace FestivalsApi;
+
+use FestivalsApi\Result\VenueSearchResult;
+use GuzzleHttp\Exception\GuzzleException;
+use IteratorAggregate;
+use LogicException;
+use Traversable;
+use function get_class;
+use function is_array;
+
+class VenueSearchIterator implements IteratorAggregate
+{
+    protected VenueSearchResult $last_result;
+
+    protected ?int $page_size = null;
+
+    protected ?array $query = null;
+
+    protected int $request_count = 0;
+
+    public function __construct(protected FestivalsApiClient $client)
+    {
+    }
+
+    /**
+     * @throws FestivalsApiClientException if API client encounters an error
+     * @throws GuzzleException if Guzzle encounters an error
+     * @throws LogicException if no query set
+     */
+    public function getIterator(): Traversable
+    {
+        $this->throwIfNoQuerySet();
+
+        $this->query['size'] = $this->page_size;
+        $this->query['from'] = 0;
```

Comment:
It may be worth pulling more of the duplicated code from EventSearchIterator into an abstract base class, and just extending that for the two specialisations? I think the only thing that differs is the code within `makeApiCall`?